msbuild/Xamarin.Mac.Tasks.Core/Tasks/DetectSdkLocations.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.MacDev.Tasks;
+using Xamarin.MacDev;
+
+namespace Xamarin.Mac.Tasks
+{
+	public class DetectSdkLocationsTaskBase : Task
+	{
+		#region Inputs
+
+		public string SessionId { get; set; }
+
+		// This is also an input
+		[Output]
+		public string SdkVersion {
+			get; set;
```

Comment:
I'm not sure why this would not build:

* Create a project for a new assembly (Xamarin.Mac.Tasks.Core)
* Move all the files you want to move there, without changing any contents.
* Reference Xamarin.Mac.Tasks.Core from where the corresponding files were previously used.

Afaict that should still work, because all you did was move code to a new assembly.

Then the next step would be to make the changes you need.